File: rb/features.rb
# YadorePublisher SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module YadorePublisherFeatures
  def self.make_feature(name)
    case name
    when "base"
      YadorePublisherBaseFeature.new
    when "test"
      YadorePublisherTestFeature.new
    else
      YadorePublisherBaseFeature.new
    end
  end
end

File: py/features.py
# YadorePublisher SDK feature factory

from feature.base_feature import YadorePublisherBaseFeature
from feature.test_feature import YadorePublisherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: YadorePublisherBaseFeature(),
        "test": lambda: YadorePublisherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
